boardswarm-client/src/device.rs

```diff
@@ -232,6 +232,14 @@ impl Device {
         Ok(())
     }
 
+    pub async fn device_info(
```

Comment:
:+1:

Reply:
the Device already monitors the info stream; so with this api you're now receiving two info streams;  The Device struct was really meant for high-level api's, so it's a bit odd to directly expose the low-level protocol like that again

Reply:
fwiw looking overall; it may make sense to push watching the device mode into the Device itself; either with a one-shot api or by exposing a stream of mode-changes. Should probably also add an api to get the current mode